routing-policy: add 'iif' and 'oif' keys to routing-policy entry

aldobrrrr opened the pull request:
## Description

It's now possible to specify the incoming interface and the outgoing interface for a routing policy entry.


Example:

```
 network:
  version: 2
  ethernets:
    if0: {}
    eth0:
      dhcp4: true
      routing-policy:
        - table: 101
          iif: eth0
        - table: 101
          oif: if0
```

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains code coverage (`make check-coverage`).
- [x] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

aldobrrrr committed:
add 'iif' key to routing-policy entry

It's now possible to specify the incoming interface for a routing
policy entry.

Example:

  network:
    version: 2
    ethernets:
      if0: {}
      eth0:
        dhcp4: true
        routing-policy:
          - table: 101
            iif: if0

Signed-off-by: Emanuele Altomare <emanuele@common-net.org>

Grissess commented:
This would close the languishing [Bug 1798417](https://bugs.launchpad.net/netplan/+bug/1798417) provided one adds a (symmetric, one hopes) implementation for `oif` as well :)

aldobrrrr committed:
add 'oif' key to routing-policy entry

It's now possible to specify the outgoing interface for a routing
policy entry.

Example:

  network:
    version: 2
    ethernets:
      if0: {}
      eth0:
        dhcp4: true
        routing-policy:
          - table: 101
            oif: if0

Signed-off-by: Emanuele Altomare <emanuele@common-net.org>

aldobrrrr commented:
> This would close the languishing [Bug 1798417](https://bugs.launchpad.net/netplan/+bug/1798417) provided one adds a (symmetric, one hopes) implementation for `oif` as well :)

As you suggested, I added the 'oif' key, thank you!

guerby commented:
@slyon looks like there's a second PR for the same feature here: https://github.com/canonical/netplan/pull/557
